### PR TITLE
Use transformers-either

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Use the `transformers-either` package if that is interesting or worthwhile.

### CODE REVIEW

File: src/Traction/Control.hs [hunk 1]
```diff
@@ -73,7 +74,7 @@ class MonadIO m => MonadDb m where
 instance MonadDb Db where
   liftDb = id
 
-instance MonadDb m => MonadDb (ExceptT e m) where
+instance MonadDb m => MonadDb (EitherT e m) where
```
Comment (markhibberd) [inline src/Traction/Control.hs]: Not sure this is right, downstream is exceptT (at least an be)

Comment (markhibberd) [reply src/Traction/Control.hs]: (I mean from a doc/error message perspective)

**Review by markhibberd — COMMENTED** [review]
Thanks. I will merge tonight, wouldn’t mind your perspective on the error message/doc thing though.

File: src/Traction/Control.hs (hunk 1 above)
Comment (tmcgilchrist) [reply src/Traction/Control.hs]: Probably should be the more general `ExceptT` type for maximum compatibility and better documentation.

If I change this back to `ExceptT` and it still works (as I expect) with `transformers-either` then I think that is a better solution.

**Comment by tmcgilchrist** [comment]
@markhibberd Switched to use `ExceptT` publicly.

**Comment by markhibberd** [comment]
This is actually merged, I just had to do a little surgery because of some other changes I had made.

**Pull request closed without merging**